File: Tests.Data/Output/HTML/Syntax Highlighting/Rust/Numbers - Input.rs

/* Topic: Integer Literals
	_____________________________________________

	Allows decimal, binary, hex, and octal.  All forms can have underscore separators.

	--- Code
	123
	123_456

	0b0110
	0b0110_1001
	0b_0110__1001

	0x01AC
	0x01AC_5BFF
	0x_01AC__5BFF

	0o1377
	0x1377_0211
	0x_1377__0211

	-123
	-123_456
	---
*/


/* Topic: Integer Literals with Suffixes
	_____________________________________________

	--- Code
	123u8
	123_456_u16

	0b0110i8
	0b0110_1001_i16

	0x01ACu32
	0x01AC_5BFF_u64
	0x_01AC__5BFF__u128

	0o1377i32
	0x1377_0211_i64
	0x_1377__0211__i128

	-123usize
	-123_456_isize
	---
*/


/* Topic: Floating Point Literals
	_____________________________________________

	Floating point literals must have a leading digit.  They cannot start with a dot like ".2".

	--- Code
	1.2
	-1.2
	+1.2

	12e3
	-12E-3
	+12E+3

	1.2e3
	-1.2E-3
	+1.2E+3
	---

	Underscores are allowed to precede the exponent digits.

	--- Code
	100_000e300_000
	+100_000.200_000e300_000
	-100_000.200_000e-_300_000
	0.200_000e+_300_000
	---
*/


/* Topic: Floating Point Literals with Suffixes
	_____________________________________________

	--- Code
	1.2f32
	-12E3f32
	+1.2E+3f64
	---
*/


/* Topic: Traps
	_____________________________________________

	--- Code
	-1
	1-1

	1-1u8
	1u8-1u8
	1u8-1

	1-1usize
	1usize-1usize
	1usize-1

	+1
	1+1

	1+1u8
	1u8+1u8
	1u8+1

	1+1usize
	1usize+1usize
	1usize+1
	---
*/

File: Tests.Data/Output/HTML/Syntax Highlighting/Python/Numbers - Input.py

# Topic: Integers
#
# --- Code
# 12
# -12
# 123456789
# -123456789
# ---

# Topic: Floating Point
#
# --- Code
# 12e1
# -12E-1
# 12.34
# -12.34
# 12.34e+5
# -12.34e-5
# .012
# -.012
# .012E+5
# -.012e10
# 0.0123
# -0.0123
# 0.0123e+5
# -0.0123E-5
# ---
#
# Ending with just a dot is allowed.
#
# --- Code
# 1.
# 123.e5
# ---

# Topic: Hex, Binary, and Octal
#
# --- Code
# 0x01230ABC
# 0Xabc12345
# 0b00100110
# 0B0110100110011100
# 0o1234
# 0O567
# ---

# Topic: Digit Separators
#
# --- Code
# 1_234_567
# 0xabc1_2345
# 0b0110_1001_1001_1100
# 1.234_567e1_2
# ---

# Topic: Imaginary Numbers
#
# Imaginary numbers are represented with a j (not an i) after the constant.
#
# --- Code
# 1.23j
# .01e-12j
# 123j
# 12.j
# ---

# Topic: Traps
#
# The fact that the hex value ends in E shouldn't make it think the following token is part of
# the number as an exponent.
#
# --- Code
# 0x000E+a
# 0x000e-1
# ---
#
# The minus sign should be part of the number for -1 but not for x-1.
#
# --- Code
# -1
# x-1
# x -1
# 1-1
# 1 -1
# (-1)
# 1 + -1
# ---
#
# Digit separators can't start a number.
#
# --- Code
# _12
# ---
#
# Numbers appearing in other contexts shouldn't be mistaken for constants.
#
# --- Code
# a1_2_3 = "45"
# ---
